File: F_Nahom_s_Array_Dilemma.py
#author: Lemi
import sys, heapq,math
from collections import defaultdict, Counter, deque
from bisect import bisect_left, bisect_right
from math import ceil, sqrt, gcd
from random import randint

rand = randint(1, 10**5)
def ran(num): return num ^ rand
def ls(): return sys.stdin.readline().split()
def ints(): return map(int, sys.stdin.readline().split())
def strs(): return map(str, sys.stdin.readline().split())
def it(): return int(sys.stdin.readline())
def st(): return sys.stdin.readline().strip()
def li(): return list(map(int, sys.stdin.readline().split()))
def yesNo(value): return 'YES' if value == 1 else 'NO' 
 


def solve():
    n = it()
    arr = li()
    
    prefix = [0] * n
    prefix[0] = arr[0]
    
    for i in range(1, n):
        prefix[i] = prefix[i - 1] + arr[i]

    q = deque()

    for i in range(n):
        while q and arr[q[-1]] <= arr[i]:
            idx = q.pop()
            new = prefix[i]
            if idx:
                new -= prefix[idx - 1]
            if new > arr[i]:
                print("NO")
                return
        q.append(i)

    arr.reverse()


    prefix = [0] * n
    prefix[0] = arr[0]
    for i in range(1, n):
        prefix[i] = prefix[i - 1] + arr[i]

    q.clear()

    for i in range(n):
        while q and arr[q[-1]] <= arr[i]:
            idx = q.pop()
            new = prefix[i]
            if idx:
                new -= prefix[idx - 1]
            if new > arr[i]:
                print("NO")
                return
        q.append(i)

    print("YES")
    
t = it()
for _ in range(t):
    solve()

File: F_Nahom_s_Array_Dilemma.cpp
from collections import deque
from sys import stdin

def I(): return int(stdin.readline().strip())
def IL(): return list(map(int, stdin.readline().strip().split()))

def solve():
    n = I()
    a = IL()
    
    pre = [0] * n
    pre[0] = a[0]
    
    # Calculating the prefix sum manually
    for i in range(1, n):
        pre[i] = pre[i - 1] + a[i]

    q = deque()

    for i in range(n):
        while q and a[q[-1]] <= a[i]:
            idx = q.pop()
            temp = pre[i]
            if idx:
                temp -= pre[idx - 1]
            if temp > a[i]:
                print("NO")
                return
        q.append(i)

    q.clear()

    for i in range(n - 1, -1, -1):
        while q and a[q[-1]] <= a[i]:
            idx = q.pop()
            temp = pre[idx]
            if i:
                temp -= pre[i - 1]
            if temp > a[i]:
                print("NO")
                return
        q.append(i)

    print("YES")
    
T = I()
for _ in range(T):
    solve()
